src/test-c-gen/c_test_file.cpp

```diff
@@ -0,0 +1,195 @@
+#include "c_test_file.h"
+
+#include <sstream>
+
+#include <test-c-gen/function_return_builder.h>
+
+/*******************************************************************\
+Function: c_test_filet::emit_standard_includes
+Purpose: Add the normal includes needed for the test harness
+\*******************************************************************/
+void c_test_filet::emit_standard_includes()
+{
+  add_line_at_root_indentation("#include <assert.h>");
+  add_line_at_root_indentation("#include <stdio.h>");
+  add_line_at_root_indentation("#include <stdlib.h>");
+  add_empty_line();
+}
+
+/*******************************************************************\
+Function: c_test_filet::emit_file_include
+Inputs:
+ file_name - The file being tested
+Purpose: To add an include for the file being tested
+\*******************************************************************/
+void c_test_filet::emit_file_include(const irep_idt &file_name)
+{
+  std::ostringstream include_line_builder;
+  include_line_builder << "#include \"";
+  include_line_builder << file_name;
+  include_line_builder << "\"";
+  add_line_at_root_indentation(include_line_builder.str());
+}
+
+/*******************************************************************\
+Function: c_test_filet::emit_main_method
+Purpose: Start creating the main method
+\*******************************************************************/
+void c_test_filet::emit_main_method()
+{
+  add_line_at_current_indentation("int test_main(int argc, char* argv)");
+  add_opening_brace(0);
+}
+
+/*******************************************************************\
+Function: c_test_filet::end_main_method
+Purpose: Put the closing brace on the main method
+\*******************************************************************/
+void c_test_filet::end_main_method()
+{
+  add_line_at_current_indentation("exit(0);");
```

Comment:
To support calling `main` from the test, I was calling the test harnesses main method `test_main` and the compiling with the flags  `-Wl,-etest_main -nostartfiles`. If you do this, then to avoid a segfault you need to call `exit(0)` at the end: http://stackoverflow.com/a/34380445